### ISSUE TITLE
Adapter initialization and updateDataSet with no notifyItemXXX()

### ISSUE THREAD

**Issue by Bylynka** [davideas/FlexibleAdapter#144]
Hi thanks for your job!

I tried to use 5.0.0-b7 for sticky headers. But it looks like it doesn't  work correctly yet(list jums when it is scrolled).
Than I tried 5.0.0-SNAPSHOT. It works correctly, but dropdown animation appears when call updateDataSet method is called.
How can I disable this animation?

**Comment by davideas** [comment]
Hi @Bylynka, thank you!
But what do you mean with _dropdown animation_?

**Comment by Bylynka** [comment]
Please look at the video [https://www.youtube.com/watch?v=da6dcHgjFrA](url)
on 'PITCHES' tab I extend FlexibleAdapter on 'TOPICS' tab I extend RecyclerView.Adapter.
On 'PITCHES' tab you can see this animation( when updateDataSet is called).

**Comment by davideas** [comment]
What you see, is the insert of the headers that is done with a notification of `notiyItemInserted()`.
showAllHeaders is also called in more parts of the adapter, to restore the initial set.

Actually you can't disable. I don't know if I can remove it, I will need to redesign some methods. But that effect is not that bad I think.

**Comment by Bylynka** [comment]
Thank you for answer! I agree with you that effect is not bad. But for current project I need list without any effects..

**Comment by davideas** [comment]
I will see what I can do, but I can't assure when it can be done.

**Comment by Bylynka** [comment]
Thank you. It will be great.

**Comment by davideas** [comment]
OK, yesterday I wanted to check something on this issue. For the moment the new code is valid only for `updateDataSet()`. Initialization works as before with the notifications.

In **experiments** branch I added the new code to skip all `notifyItemXXX()` only when animate is set `false`. Just before calling `notifyDataSetChanged()`, I complete the adapter set with expandables expanded and headers to be shown. Which means no effects at all when new items are fetched or items are removed from the list.

This can be obtained by calling `updateDataSet(items)` or `updateDataSet(items, false)`. This was already in place but subItems and headers were applied after the `notifyDataSetChanged()`. Now they are added before.
In the demoApp I added a checkMenuBox to see the difference when `animate = true` or `false`:
- When `true` it synchronizes all new and existent items with `notifyItemXXX()`.
- When `false` it doesn't make any animation, because `notifyDataSetChanged()` is called.
